compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -4791,6 +4791,7 @@ private void validateResourceFunctionAttachedToObject(BLangFunction funcNode, BO
             return;
         }
         funcNode.symbol.flags |= Flags.RESOURCE;
+        funcNode.symbol.flags |= Flags.PUBLIC;
```

Comment:
Made resource functions implicitly public as the remote functions. Otherwise, we cannot access resource functions from other modules. This is a breaking change since any attempt to expose non-public symbol from a resource function will result in a compilation error @MaryamZi @KavinduZoysa @pcnfernando @gimantha Please verify this.

Reply:
Verified offline